### PR TITLE
Export de-components Lib and Minor refactor. 

### PR DESCRIPTION
**Pull request by sri2k1us**
- As part of my experiments, I also added a `Rate` component.
See https://github.com/sri2k1us/de-components/pull/1 for more info.

### CODE REVIEW

File: package.json
Comment (psarando) [inline package.json:14]: I wonder if there's a way to accomplish this with just `react-scripts build`? Maybe if everything is still exported from `src/index.js`? 🤔 

I can try experimenting with it before or after this PR is merged.

File: src/components/announcer/CyverseAnnouncer.js
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: 👍

File: src/components/announcer/CyverseAnnouncer.js
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: 👍

File: src/components/announcer/CyverseAnnouncer.js
Comment (psarando) [inline src/components/announcer/CyverseAnnouncer.js]: We should complete this documentation by further documenting this object:
```javascript
/**
 * @typedef {(TOP|BOTTOM)} VerticalAlignmentEnum
 */

/**
 * @typedef {(RIGHT|LEFT|CENTER)} HorizontalAlignmentEnum
 */

/**
 * @typedef {(INFO|ERROR|WARNING|SUCCESS)} VariantEnum
 */

/**
 * @typedef {Object} QueueMessage
 * @property {string} text - The message text.
 * @property {VariantEnum} variant - The message variant.
 * @property {number} duration - The message duration.
 * @property {HorizontalAlignmentEnum} horizontal - The message horizontal alignment.
 * @property {VerticalAlignmentEnum} vertical - The message vertical alignment.
 */

/**
 *  Queue messages needed to be announced using CyverseAnnouncer
 *  @param {QueueMessage} msg - A message with configuration.
 */
```

Then above, these enum constants can also be documented:
```javascript
/**
 * A message variant enum that will display info messages.
 * @constant
 */
const INFO = "info";
```

File: src/components/lib.js
Comment (psarando) [inline src/components/lib.js]: They way we have the `build` script and this export set up now, I think we would have to use something like the following in our main UI code to use these components:
```javascript
import { CyVerseAnnouncer, announce, Rate } from "@cyverse-de/de-components";
```

I wonder if we should imitate a package structure similar to [material-ui](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/index.js) (and we should decide soon before these components are imported in too many places)?

For example, to use a MUI `Button` and an icon:
```javascript
import Button from "@material-ui/core/Button";
import HelpIcon from "@material-ui/icons/Help";
```

Would it be more convenient to use imports like the following for our components?
```javascript
import CyVerseAnnouncer from "@cyverse-de/de-components/CyVerseAnnouncer";
import Rate from "@cyverse-de/de-components/Rate";
import SomethingSoon from "@cyverse-de/de-components/future-sub-package/SomethingSoon";
```

Basically treating `@cyverse-de/de-components` the same as `@material-ui/core`, but sub-packaging from there rather than creating separate packages (like `@material-ui/icons`).

I don't know the best way to organize the exports to accomplish this right now, but we could experiment and figure something out if we wanted to go this way.

File: src/components/lib.js
Comment (psarando) [inline src/components/lib.js]: I think `CyVerse` should be properly cased, so at least this import should be renamed `CyVerseAnnouncer`, but the file and classes in `CyverseAnnouncer.js` should probably get renamed to `CyVerseAnnouncer` as well.

File: .gitignore
Comment (psarando) [inline .gitignore]: Since we already ignore `/dist` above, we don't need this `dist/lib.js` entry too.

File: src/components/rating/Rate.js
Comment (psarando) [inline src/components/rating/Rate.js]: Looks like one of these `span` tags is at the wrong indentation level.

**Review by psarando — CHANGES_REQUESTED** [review]
Looks good 👍 

I have some questions on the library organization, and a few suggestions, for docs, branding, etc.

File: package.json
Comment (psarando) [reply package.json:14]: Oh wait, never mind that. Since `react-scripts build` generates a `build` folder with `index.html` and assets under `build/static/js` that have cache-busting names. So that won't work for library components.

File: src/components/lib.js
Comment (ash3rz) [reply src/components/lib.js]: I think it'd be more convenient to have the imports like how MUI has them, but also not sure how much work it would take to replicate MUI's package structure.

**Review by ash3rz — COMMENTED** [review]
I had the same feeling about using proper casing for `CyVerse` that Paul mentioned, but other than that, LGTM 👍

File: src/components/announcer/CyverseAnnouncer.js
Comment (sri2k1us) [reply src/components/announcer/CyverseAnnouncer.js]: Great! I didn't knew how to document Objects. Thanks

File: src/components/lib.js
Comment (sri2k1us) [reply src/components/lib.js]: Sounds good. I will rename them.

File: .gitignore
Comment (sri2k1us) [reply .gitignore]: 👍

File: src/components/rating/Rate.js
Comment (sri2k1us) [reply src/components/rating/Rate.js]: Prettier - Please take care of it 😉

**Comment by sri2k1us** [comment]
I have run `prettier` once in this branch.

File: src/App.js
Comment (psarando) [inline src/App.js]: LGTM, :shipit: once you delete these extra spaces 👾

**Review by psarando — APPROVED** [review]

**Comment by sri2k1us** [comment]
I updated version in `package.json` with `npm version minor -m 'Upgrade to %s'`.

**Comment by sri2k1us** [comment]
Thanks for the reviews!

**Pull request merged**
